python/tchannel/tornado/message_builder.py

```diff
@@ -0,0 +1,65 @@
+from ..messages import Types
+
+
+class MessageBuilder(object):
+
+    ARGS = ["arg_1", "arg_2", "arg_3"]
+
+    def __init__(self):
+        """key: message_id
+           value: incomplete messages
+        """
+        self.message_buffer = {}
+
+    @staticmethod
+    def _get_cur_pending_arg(message):
+        cur_arg = 0
+        if message.arg_3 != "":
+            cur_arg = 2
+        elif message.arg_2 != "":
+            cur_arg = 1
+        print "cur_arg" + str(cur_arg)
+        return cur_arg
+
+    def compose_map(self, call_msg):
+        cur_arg = self._get_cur_pending_arg(call_msg)
+
+        arg_map = {"arg_1": "",
+                   "arg_2": "",
+                   "arg_3": ""}
+        t = 0
+
+        for i in range(cur_arg, 3):
+            arg_map[self.ARGS[i]] = self.ARGS[t]
+            t += 1
+
+        return arg_map
+
+    def build(self, message_id, message):
+
+        if message.message_type is Types.CALL_REQ or\
+           message.message_type is Types.CALL_RES:
+            if message.flags == 0x01:  # streaming message
```

Comment:
Seems like these values (`0x01`, `0x00`, etc...) should be in constants so that we can 1 change them and 2 reuse them - @abhinav @blampe waht do you guys think?